### PR TITLE
feat: add new moderation apis

### PR DESCRIPTION
**Pull request by peterdeme**
New Moderation API

### CODE REVIEW

File: query.go [hunk 1]
```diff
@@ -217,3 +234,14 @@ func (c *Client) QueryMessageFlags(ctx context.Context, q *QueryOption) (*QueryM
 	err = c.makeRequest(ctx, http.MethodGet, "moderation/flags/message", values, nil, &resp)
 	return &resp, err
 }
+
+type QueryFlagReportsResponse struct {
+	Response
+	FlagReports []*FlagReport `json:"flag_reports"`
+}
+
+func (c *Client) QueryFlagReportsAsync(ctx context.Context, q *QueryFlagReportsRequest) (*QueryFlagReportsResponse, error) {
```
Comment (yaziine) [inline query.go:243]: why `Async` ?

Comment (peterdeme) [reply query.go:243]: by mistake, sorry. disadvantage of working on 6 languages :)

Comment (yaziine) [reply query.go]: haha no worries, all credits go to you 😊

**Review by yaziine — APPROVED** [review]
lgtm

**Pull request merged**